tests/Microsoft.Identity.Web.Test/L1L2CacheTests.cs

```diff
@@ -202,20 +206,56 @@ public async Task EmptyL1Cache_ReadL2AndSetL1_TestAsync()
             // Arrange
             byte[] cache = new byte[3];
             cache[0] = 4;
+            TelemetryData telemetryData = new TelemetryData();
             AssertCacheValues(_testCacheAdapter);
             _testCacheAdapter._distributedCache.Set(DefaultCacheKey, cache);
             Assert.Single(L2Cache._dict);
             Assert.NotNull(_testCacheAdapter._memoryCache);
             Assert.Equal(0, _testCacheAdapter._memoryCache.Count);
 
             // Act
-            byte[]? result = await _testCacheAdapter.TestReadCacheBytesAsync(DefaultCacheKey).ConfigureAwait(false);
+            byte[]? result = await _testCacheAdapter.TestReadCacheBytesAsync(DefaultCacheKey, telemetryData).ConfigureAwait(false);
 
             // Assert
             Assert.NotNull(result);
             Assert.Equal(4, result[0]);
             Assert.Equal(1, _testCacheAdapter._memoryCache.Count);
             Assert.Single(L2Cache._dict);
+            Assert.Equal(CacheLevel.L2Cache, telemetryData.CacheLevel);
+        }
+
+        [Fact]
```

Comment:
These tests are ok, but higher level tests are always better because they are easier to read and cover the E2E. Have a look at `CacheExtensionsTests`. 

It should be pretty easy to hook up MSAL to a setup with L1+L2 cache and to a setup with only L1 cache. We should make sure we have a test for when L1 cache is disabled (this is one of the options).